src/sb_handle/kernel_constructor.hpp

```diff
@@ -255,5 +270,58 @@ static SYCL_BLAS_INLINE cl::sycl::event execute_tree(queue_t q_,
     return ev;
   }
 }
+
+template <int using_local_memory, typename queue_t, typename expression_tree_t>
```

Comment:
@muhammad-tanvir-1211  why the kernel constructor requires to change? This file will construct the expression tree kernel irrespective of what the expression is. The tree has nothing to do with a specific operation and it just pass whatever is the expression. This file should not be touched at all. If you have anything requires specific execution rule, it should go to the `SB_Handle` class or gemm launcher class not here

Reply:
The change in kernel_constructor* is required because we want to use 2D nd-range instead of 1D. For the 2D range, we need the size of global and local range for both dimensions and then explicitly specify the nd-range before calling q.submit(). 

The eval functions also need to change since the existing functions are all hardcoded to accept 1D nd-range whereas my implementation requires a 2D nd-range.